.github/workflows/upload-and-deploy.yaml

```diff
@@ -40,12 +40,12 @@ jobs:
         with:
           version: 1.37.0
 
-      - name: Set namespace for branch
+      - name: Set namespace for non-default branch or for tag
         if: github.ref_name != 'main'
         run: echo "NAMESPACE=$GITHUB_REF_NAME" >> $GITHUB_ENV
 
       - name: Copy files to templates bucket
-        run: python src/scripts/manage_artifacts/artifacts.py --upload --namespace ${{ env.NAMESPACE }}
+        run: python src/scripts/manage_artifacts/artifacts.py --upload --ref $GITHUB_REF_NAME
```

Comment:
So namespaces are being applied for templates but refs are being used for artifacts/paths